File: src/main/scala/com/teambr/modularsystems/core/client/itemtooltip/ToolTipEvent.scala
package com.teambr.modularsystems.core.client.itemtooltip

import net.minecraft.block.Block
import net.minecraftforge.event.entity.player.ItemTooltipEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

/**
 * This file was created for the Modular-Systems
 *
 * Modular-Systems if licensed under the is licensed under a
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since August 02, 2015
 */
class ToolTipEvent {
    @SubscribeEvent def onToolTip(event: ItemTooltipEvent) {
        var item: IItemTooltip = null

        //Extract the item/block
        Block.getBlockFromItem(event.itemStack.getItem) match {
            case isABlock: IItemTooltip => item = isABlock
            case _ => event.itemStack.getItem match {
                case theItem: IItemTooltip => item = theItem
                case _ =>
            }
        }

        //Add the tip should it need
        if (item != null) {
            val tips: java.util.List[String] = item.returnTooltip
            if (tips != null) for (tip <- tips) event.toolTip.add(tip)
        }
    }
}

File: src/main/scala/com/teambr/modularsystems/storage/event/ToolTipEvent.java
package com.teambr.modularsystems.storage.event;

import com.teambr.bookshelf.client.gui.GuiColor;
import com.teambr.modularsystems.storage.container.ContainerStorageCore;
import net.minecraft.client.Minecraft;
import net.minecraftforge.event.entity.player.ItemTooltipEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

import java.text.NumberFormat;
import java.util.Locale;

/**
 * This file was created for Modular-Systems
 * <p/>
 * Modular-Systems is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis "pauljoda"
 * @since 3/28/2016
 */
public class ToolTipEvent {
    @SubscribeEvent
    public void onToolTip(ItemTooltipEvent event) {
        if(Minecraft.getMinecraft().thePlayer.openContainer != null &&
                Minecraft.getMinecraft().thePlayer.openContainer instanceof ContainerStorageCore && event.getItemStack().stackSize > 1000) {
            event.getToolTip().add(GuiColor.YELLOW + "Quantity");
            event.getToolTip().add("  " + NumberFormat.getNumberInstance(
                    Locale.forLanguageTag(Minecraft.getMinecraft().gameSettings.language)).format(event.getItemStack().stackSize));
            event.getToolTip().add(GuiColor.YELLOW + "Stacks");
            event.getToolTip().add("  " +  NumberFormat.getNumberInstance(
                    Locale.forLanguageTag(Minecraft.getMinecraft().gameSettings.language)).format(
                    Math.floor(event.getItemStack().stackSize / event.getItemStack().getMaxStackSize())) +
                    " + " + event.getItemStack().stackSize % event.getItemStack().getMaxStackSize());
        }
    }
}
